core/src/main/scala/org/apache/spark/sql/hive/SnappyHiveExternalCatalog.scala

```diff
@@ -77,13 +81,19 @@ class SnappyHiveExternalCatalog private[hive](val conf: SparkConf,
   protected val cachedCatalogTables: LoadingCache[(String, String), CatalogTable] = {
     val cacheLoader = new CacheLoader[(String, String), CatalogTable]() {
       override def load(name: (String, String)): CatalogTable = {
-        logDebug(s"Looking up data source for ${name._1}.${name._2}")
-        withHiveExceptionHandling(SnappyHiveExternalCatalog.super.getTableOption(
-          name._1, name._2)) match {
-          case None =>
-            nonExistentTables.put(name, java.lang.Boolean.TRUE)
-            throw new TableNotFoundException(name._1, name._2)
-          case Some(catalogTable) => finalizeCatalogTable(catalogTable)
+        try {
+          withHiveExceptionHandling(SnappyHiveExternalCatalog.super.getTableOption(
+            name._1, name._2)) match {
+            case None =>
+              nonExistentTables.put(name, java.lang.Boolean.TRUE)
+              throw new TableNotFoundException(name._1, name._2)
+            case Some(catalogTable) => finalizeCatalogTable(catalogTable)
+          }
+        } catch {
+          case _: NullPointerException =>
+            throw new AnalysisException(
+              s"Table ${name._1}.${name._2} might be inconsistent in hive catalog. " +
+                  "refer to troubleshooting section of documentation for resolution")
```

Comment:
Modify the message  to say some thing like-
s"Table entry ${name._1}.${name._2} might be inconsistent in hive catalog.  Use system procedure SYS.REMOVE_METASTORE_ENTRY to remove inconsistency."
+  "Refer to troubleshooting section of documentation for more details"